File: Java/versandetiketten/src/Integratoren/VersandEtikettenErsteller.java
package Integratoren;

import java.util.*;

import Daten.*;
import IO.MiddleWareInterface;
import Operationen.*; 

//---------------------------------------------------------------------------------
// Integrator, der die Versandetiketten erstellen lässt
//i/ Dieser Integrator enthält weder Logik noch eine Schleife!
//---------------------------------------------------------------------------------
public class VersandEtikettenErsteller {
	
	private int letzteEtikettNr = 0;
	
	public int getEtikettZaehler() {
		return this.letzteEtikettNr;
	}
	
	public MiddleWareInterface middleWare = IoFactory.getMiddleWare();
	
	public EtikettVariante1 versandEtikettVariante1(String kundeID, String adresseID, String laenderKennung, String weitereID) throws Exception {
		EtikettVariante1 erstelltesEtikett = this.erzeugeNeuesEtikett();
		
		erstelltesEtikett.anschriftZeilen = this.anschriftErstellung(kundeID, adresseID, laenderKennung);
		
		return erstelltesEtikett;
	}

	private EtikettVariante1 erzeugeNeuesEtikett() {
		EtikettVariante1 erstelltesEtikett = new EtikettVariante1();
		
		erstelltesEtikett.EtikettNr = ++this.letzteEtikettNr;
		
		return erstelltesEtikett;
	}
	
	public List<String> anschriftErstellung(String kundeID, String adresseID, String laenderKennung) throws Exception {
		Kunde kunde = middleWare.leseKunde(kundeID);
		Adresse adresse = middleWare.leseAdresse(adresseID);
		
		IAnschriftFormatierer anschriftFormatierer = Integratoren.OperatorFactory.buildAnschriftFormatierer(laenderKennung);
		
		return anschriftFormatierer.AnschriftErstellung(kunde, adresse);
	}
}

File: Csharp/VersandEtiketten/Integration/VersandEtikettenErsteller.cs
using Daten;
using IO_Operationen;
using Operationen;

namespace Integration
{
    public class VersandEtikettenErsteller
    {
        public int EtikettZaehler { get; private set; }

        public MiddleWareInterface middleWare = IoFactory.getMiddleWare();

        public EtikettVariante1 VersandEtikettVariante1(String? kundeID, String? adresseID, String? laenderKennung, String? weitereID)
        {
            EtikettVariante1 erstelltesEtikett = this.ErzeugeNeuesEtikett();
		
		erstelltesEtikett.AnschriftZeilen = this.AnschriftErstellung(kundeID, adresseID, laenderKennung);
		
		return erstelltesEtikett;
	}

    private EtikettVariante1 ErzeugeNeuesEtikett()
    {
        EtikettVariante1 erstelltesEtikett = new EtikettVariante1();

        erstelltesEtikett.EtikettNr = ++this.EtikettZaehler;

        return erstelltesEtikett;
    }

    public List<String?>? AnschriftErstellung(String? kundeID, String? adresseID, String? laenderKennung)
    {
        Kunde kunde = middleWare.LeseKunde(kundeID);
        Adresse adresse = middleWare.LeseAdresse(adresseID);

        IAnschriftFormatierer anschriftFormatierer = OperatorFactory.buildAnschriftFormatierer(laenderKennung);
		
		return anschriftFormatierer.AnschriftErstellung(kunde, adresse);
    }
}
}